vmm/src/memory_manager.rs

```diff
@@ -594,12 +642,16 @@ impl MemoryManager {
         // Start address needs to be non-contiguous with last memory added (leaving a gap of 256MiB)
         // and also aligned to 128MiB boundary. It must also start at the 64bit start.
         let mem_end = self.guest_memory.memory().last_addr();
+        #[cfg(target_arch = "x86_64")]
         let start_addr = if mem_end < arch::layout::MEM_32BIT_RESERVED_START {
             arch::layout::RAM_64BIT_START
         } else {
             GuestAddress((mem_end.0 + 1 + (256 << 20)) & !((128 << 20) - 1))
         };
 
+        #[cfg(target_arch = "aarch64")]
+        let start_addr = GuestAddress((mem_end.0 + 1 + (256 << 20)) & !((128 << 20) - 1));
```

Comment:
Refactor out into a `start_addr()` function.

Reply:
I refactored more to cover the similar calculation for the start address of device space in `new()`.